src/devices/src/virtio/block/event_handler.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use std::os::unix::io::AsRawFd;
+
+use polly::epoll::{EpollEvent, EventSet};
+use polly::event_manager::{EventManager, Subscriber};
+
+use crate::virtio::block::device::Block;
+use crate::virtio::VirtioDevice;
+
+impl Subscriber for Block {
+    // Handle an event for queue or rate limiter.
+    fn process(&mut self, event: &EpollEvent, _: &mut EventManager) {
+        if !self.is_activated() {
+            warn!("The device is not yet activated. Events can not be handled.");
+            return;
+        }
+
+        let queue_evt = self.queue_evt.as_raw_fd();
+        let rate_limiter_evt = self.rate_limiter.as_raw_fd();
+
+        let source = event.fd();
+        let event_set = event.event_set();
+
+        // TODO: also check for errors. Pending high level discussions on how we want
```

Comment:
I saw this comment before. Is there a tracking issue?

Reply:
#1627